src/data/chats/chat.rs

```diff
@@ -71,9 +125,36 @@ impl Chat {
             messages_update_receiver: rx,
             is_streaming: false,
             title_state: TitleState::default(),
+            chats_dir,
         }
     }
 
+    pub fn load(path: PathBuf, chats_dir: PathBuf) -> Result<Self> {
+        let (tx, rx) = channel();
+
+        match read_from_file(path) {
+            Ok(json) => {
+                let mut chat: Chat = serde_json::from_str(&json)?;
+                chat.messages_update_sender = tx;
+                chat.messages_update_receiver = rx;
+                chat.chats_dir = chats_dir;
+                Ok(chat)
+            }
+            Err(_) => Err(anyhow!("Couldn't read chat file from path")),
+        }
+    }
+
+    pub fn save(&self) {
+        let json = serde_json::to_string(&self).unwrap();
+        let path = self.chats_dir.join(self.file_name());
+        write_to_file(path, &json).unwrap();
+    }
+
+    // TODO: Make a field
+    fn file_name(&self) -> String {
```

Comment:
Nit: Why would make a field if this is just a computed value? Maybe we can just put this calculation inline in the `save` function?

Reply:
oh i never did this, its probably not that big of a deal, but this creates a new String on each save rn

Reply:
I see..